testsuite/features/step_definitions/navigation_steps.rb

```diff
@@ -85,23 +85,33 @@
   end
 end
 
+$max_completed_aid = 0
```

Comment:
I don't like this state machine approach, with a global variable. It's complicated and I have a feeling it could be fragile.

Why can't we simply follow @srbarrios 's suggestion? If a race really occurs, then we can reconsider.

Reply:
I think that handling the events using a cache of processed Ids could be a good approach, I would not use a integer, but a list of integers for all processed events, as otherwise that might conflict if we try to parallelize our tests.

**BUT**, we can't afford this refactor now, now we only can fix stuff in the simple way we can or keep it failing until GM.
I don't want to risk breaking other scenarios as this refactor don't only affects the test to reboot a terminal, but will affect any other scenario checking for events.

Said that, if we find an easier approach now, even if it could have some corner case, I would prefer it, and then we can keep this PR open to apply this refactor after well discuss it other squad members. But in overall, I like the idea of handling our events by ID, as any other ways we always had issues.